src/System.Net.Sockets/src/System/Net/Sockets/AcceptOverlappedAsyncResult.Windows.cs

```diff
@@ -101,8 +96,6 @@ internal void SetUnmanagedStructures(byte[] buffer, int addressBufferLength)
 
         private void LogBuffer(long size)
         {
-            if (!NetEventSource.IsEnabled) return;
```

Comment:
Technically it would be a bad assert, in that even if the caller checks that tracing is enabled, it could be disabled between that call and the assert.  That's why I've been hesitant to add such asserts. In practice, it would be unlikely to occur, in particular since the asserts would be debug-build only.